app/controllers/shipping_controller.rb

```diff
@@ -0,0 +1,43 @@
+require 'active_shipping'
+
+class ShippingController < ApplicationController
+  skip_before_filter :verify_authenticity_token
+
+  def ups_rates
+    ups = ActiveShipping::UPS.new(login: ENV['UPS_LOGIN'], password: ENV['UPS_PASSWORD'], key: ENV['UPS_ACCESS_KEY'])
+    ups_rates = get_rates(ups)
+    render :json => ups_rates, :status => (ups_rates.flatten.include?("error") ? 204 : 200)
+  end
+
+  def usps_rates
+    usps = ActiveShipping::USPS.new(login: ENV['ACTIVESHIPPING_USPS_LOGIN'])
+    usps_rates = get_rates(usps)
+    render :json => usps_rates, :status => (usps_rates.flatten.include?("error") ? 204 : 200)
+  end
+
+  private
+    def get_rates(carrier)
+      destination = location(params["destination"])
+      origin = location(params["origin"])
+      package = package(params["package"])
+      begin
+        response = carrier.find_rates(origin, destination, package)
+        return response.rates.sort_by(&:price).collect {|rate| [rate.service_name, rate.price]}
```

Comment:
It's generally not recommended to chain methods together in case one of them fails. It's just easier code to read and single responsibility that we learned about from Sandi Metz' POOTR.